typedData/typedData_test.go

```diff
@@ -0,0 +1,346 @@
+		},
+		{
+			TypedData:           typedDataExamples["example_enum"],
+			Address:             "0xCD2a3d9F938E13CD947Ec05AbC7FE734Df8DD826",
+			ExpectedMessageHash: "0x416b85b18063b1b3420ab709e9d5e35cb716691d397c5841ce7c5198ee30bf",
+		},
+		{
+			TypedData:           typedDataExamples["allInOne"],
+			Address:             "0xCD2a3d9F938E13CD947Ec05AbC7FE734Df8DD826",
+			ExpectedMessageHash: "0x300dc63cc85a15529bba5ed482009be716645fa9f2c64bd8716cf6f34767651",
+		},
+	}
+
+	for _, test := range testSet {
+		hash, err := test.TypedData.GetMessageHash(test.Address)
+		require.NoError(t, err)
+
+		require.Equal(t, test.ExpectedMessageHash, hash.String())
+	}
+}
+
+// BenchmarkGetMessageHash is a benchmark function for testing the GetMessageHash function.
+//
+// It tests the performance of the GetMessageHash function by running it with different input sizes.
+// The input size is determined by the bit length of the address parameter, which is converted from
+// a hexadecimal string to a big integer using the HexToBN function from the utils package.
+//
+// Parameters:
+// - b: a testing.B object that provides methods for benchmarking the function
+// Returns:
+//
+//	none
+func BenchmarkGetMessageHash(b *testing.B) {
+	ttd := BMockTypedData(b)
+
+	addr := "0xCD2a3d9F938E13CD947Ec05AbC7FE734Df8DD826"
+	b.Run(fmt.Sprintf("input_size_%d", len(addr)), func(b *testing.B) {
+		result, err := ttd.GetMessageHash(addr)
+		require.NoError(b, err)
+		require.NotEmpty(b, result)
+	})
+}
+
+// TestGeneral_GetTypeHash tests the GetTypeHash function.
+//
+// It tests the GetTypeHash function by calling it with different input values
+// and comparing the result with expected values. It also checks that the
+// encoding of the types matches the expected values.
+//
+// Parameters:
+// - t: The testing.T object used for reporting test failures and logging test output
+// Returns:
+//
+//	none
+func TestGetTypeHash(t *testing.T) {
+	type testSetType struct {
+		TypedData    TypedData
+		TypeName     string
+		ExpectedHash string
+	}
+	testSet := []testSetType{
+		{
+			TypedData:    typedDataExamples["baseExample"],
+			TypeName:     "StarkNetDomain",
+			ExpectedHash: "0x1bfc207425a47a5dfa1a50a4f5241203f50624ca5fdf5e18755765416b8e288",
+		},
+		{
+			TypedData:    typedDataExamples["baseExample"],
+			TypeName:     "Mail",
+			ExpectedHash: "0x13d89452df9512bf750f539ba3001b945576243288137ddb6c788457d4b2f79",
+		},
+		{
+			TypedData:    typedDataExamples["example_baseTypes"],
+			TypeName:     "Example",
+			ExpectedHash: "0x1f94cd0be8b4097a41486170fdf09a4cd23aefbc74bb2344718562994c2c111",
+		},
+		{
+			TypedData:    typedDataExamples["example_presetTypes"],
+			TypeName:     "Example",
+			ExpectedHash: "0x1a25a8bb84b761090b1fadaebe762c4b679b0d8883d2bedda695ea340839a55",
+		},
+		{
+			TypedData:    typedDataExamples["session_MerkleTree"],
+			TypeName:     "Session",
+			ExpectedHash: "0x1aa0e1c56b45cf06a54534fa1707c54e520b842feb21d03b7deddb6f1e340c",
```

Comment:
where did this come from?

Reply:
Explained in my previous comment
https://github.com/NethermindEth/starknet.go/pull/637#discussion_r1879436991